src/mon/Monitor.cc

```diff
@@ -2103,16 +2104,25 @@ void Monitor::handle_probe_reply(MonOpRequestRef op)
              << " vs my version " << paxos->get_version()
              << " (ok)"
              << dendl;
-
-    if (monmap->contains(name) &&
-        !monmap->get_addrs(name).front().is_blank_ip()) {
+    bool in_map = false;
+    const auto& my_info = monmap->mon_info.find(name);
+    const map<string,string> *map_crush_loc;
```

Comment:
Initialize to nullptr.  Also, can you simply use map_crush_loc being non-null to indicate in_map and remove the second variable?